tfx/examples/cifar10/cifar10_utils_native_keras.py

```diff
@@ -0,0 +1,291 @@
+
+  Returns:
+    The image classifcation Keras Model and the backbone MobileNet model
+  """
+  # We create a MobileNet model with weights pre-trained on ImageNet.
+  # We remove the top classification layer of the MobileNet, which was
+  # used for classifying ImageNet objects. We will add our own classification
+  # layer for CIFAR10 later. We use average pooling at the last convolution
+  # layer to get a 1D vector for classifcation, which is consistent with the
+  # origin MobileNet setup
+  base_model = tf.keras.applications.MobileNet(
+      input_shape=(224, 224, 3), include_top=False, weights='imagenet',
+      pooling='avg')
+
+  # We add a Dropout layer at the top of MobileNet backbone we just created to prevent
+  # overfiting, and then a Dense layer to classifying CIFAR10 objects
+  model = tf.keras.Sequential([
+      tf.keras.layers.InputLayer(
+          input_shape=(224, 224, 3), name=transformed_name(IMAGE_KEY)),
+      base_model,
+      tf.keras.layers.Dropout(0.3),
+      tf.keras.layers.Dense(10, activation='softmax')
+  ])
+
+  return model, base_model
+
+def _augment_image(image_feature):
+  """Perform data augmentation on image features.
+
+  Args:
+    image_feature: The image feature.
+
+  Returns:
+    Augmented image feature.
+  """
+  image_feature = tf.image.random_flip_left_right(image_feature)
+  image_feature = tf.image.resize_with_crop_or_pad(image_feature, 250, 250)
+  image_feature = tf.image.random_crop(image_feature, (224, 224, 3))
+  return image_feature
+
+def _data_augmentation(is_train, image_feature):
+  """Perform data augmentation on data.
+
+  Args:
+    is_train: Whether the data comes from train set.
+    image_feature: The image feature.
+
+  Returns:
+    Map from string feature key to transformed feature operations.
+  """
+  image_feature = tf.cond(tf.equal(is_train, 1), lambda: _augment_image(image_feature),
+                          lambda: image_feature)
+  return is_train, image_feature
+
+# TFX Transform will call this function.
+def preprocessing_fn(inputs):
+  """tf.transform's callback function for preprocessing inputs.
+
+  Args:
+    inputs: map from feature keys to raw not-yet-transformed features.
+
+  Returns:
+    Map from string feature key to transformed feature operations.
+  """
+  outputs = {}
+
+  image_features = tf.map_fn(lambda x: tf.io.decode_png(x[0], channels=3),
+                             inputs[IMAGE_KEY], dtype=tf.uint8)
+  image_features = tf.cast(image_features, tf.float32)
+  image_features = tf.image.resize(image_features, [224, 224])
+  image_features = tf.keras.applications.mobilenet. \
+                         preprocess_input(image_features)
+
+  # Map function has to return a tuple when input is a tuple
+  # we are only interested in the augmented features
+  _, image_features = tf.map_fn(lambda x: _data_augmentation(x[0], x[1]),
+                                (inputs[SPLIT_KEY], image_features))
+
```

Comment:
lets add a tf.ensure_shape() right here on image_features with the expected size.